### PR TITLE
Adding a section about configuration files

### PR DESCRIPTION
**Pull request by spacekookie**
This is probably the most WIP right now.

It mentions `confy` directly and has some work left to do for discovering, promoting or finishing other crates

### CODE REVIEW

File: src/in-depth/config-files.md [hunk 1]
```diff
@@ -1 +1,43 @@
 # Using config files
+
+Dealing with configurations can be annoying
+especially if you support multiple operating systems
+which all have their own places
+for short and long-term files.
```
Comment (killercup) [inline src/in-depth/config-files.md]: > for short[-] and long-term files.

This how English works, too, yes?

File: src/in-depth/config-files.md [hunk 2]
```diff
@@ -1 +1,43 @@
...
+
+There are multiple solutions to this,
+some being more low-level than others.
+
+The easiest to use crate is `confy`
```
Comment (killercup) [inline src/in-depth/config-files.md]: > The {++crate++} easiest to use {--crate--} is called `confy`

I'd also end the sentence here

File: src/in-depth/config-files.md [hunk 3]
```diff
@@ -1 +1,43 @@
...
+```
+
+This is incredibly easy to use
+for which you of course surrender configurability.
+But if a simple config is all you want,
+this crate might be for you!
+
+## Configuration environments
+
+**TODO**
```
Comment (killercup) [inline src/in-depth/config-files.md]: pls use `<aside class="todo">` as in other files

File: src/in-depth/config-files.md (hunk 1 above)
Comment (spacekookie) [reply src/in-depth/config-files.md]: Probably :sweat_smile:

**Comment by killercup** [comment]
thanks <3

**Pull request merged**